packages/overlay/stories/overlay.stories.ts

```diff
@@ -1413,3 +1413,72 @@ export const virtualElement = (args: Properties): TemplateResult => {
 virtualElement.args = {
     placement: 'right-start' as Placement,
 };
+
+export const virtualElementDeclaratively = (
+    args: Properties
+): TemplateResult => {
+    const handleContextmenu = async (event: PointerEvent): Promise<void> => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        const overlay = document.querySelector('#overlay') as Overlay;
+
+        const virtualTrigger = new VirtualTrigger(event.clientX, event.clientY);
+
+        overlay.triggerElement = virtualTrigger;
+        overlay.placement = args.placement;
+        overlay.offset = 0;
+        overlay.type = 'auto';
+
+        overlay.open = true;
+        overlay.willPreventClose = true;
+    };
+    return html`
+        <style>
+            .app-root {
+                position: absolute;
+                inset: 0;
+            }
+        </style>
+        <div
+            class="app-root"
+            @contextmenu=${{
+                capture: true,
+                handleEvent: handleContextmenu,
+            }}
+        >
+            <sp-overlay id="overlay">
+                <sp-popover
+                    style="width:300px;"
+                    @click=${(event: PointerEvent) => {
+                        if (
+                            (event.target as HTMLElement).localName ===
+                            'sp-menu-item'
+                        ) {
+                            event.target?.dispatchEvent(
+                                new Event('close', { bubbles: true })
+                            );
+                        }
+                    }}
```

Comment:
I see this is adopted from the other story. I can't remember why I leveraged this, rather than the `change` event on the `<sp-menu>` which already does the element checking for us. Should be able to use:
```
<sp-menu
    @change=${() => {
        event.target?.dispatchEvent(
            new Event('close', { bubbles: true })
        );
    }}
```
Instead.